pink/barriers/barrier.py

```diff
@@ -172,15 +179,23 @@ def compute_qp_objective(
             Tuple containing the quadratic objective matrix (H) and linear
                 objective vector (c).
         """
-        jac = self.compute_jacobian(configuration)
+        if (
+            self.__q_cache is None
+            or self.__jac_cache is None
+            or self.__barrier_cache is None
+        ) or not np.allclose(self.__q_cache, configuration.q):
+            self.__q_cache = configuration.q.copy()
+            self.__jac_cache = self.compute_jacobian(configuration).copy()
+            self.__barrier_cache = self.compute_barrier(configuration).copy()
+
         H = np.zeros((configuration.model.nv, configuration.model.nv))
         c = np.zeros(configuration.model.nv)
 
         if self.safe_displacement_gain > 1e-6:
             self.safe_displacement = self.compute_safe_displacement(
                 configuration
             )
-            jac_squared_norm = np.linalg.norm(jac) ** 2
+            jac_squared_norm = np.sum(self.__jac_cache**2)
```

Comment:
Indeed, this is a surprise for me, it becomes even worse when dimensions grow. 

Thank you very much for noting this, I would be more careful about such reasoning in the future.

Reply:
This is very surprising for me that it works in this way.

Thank you for your attention to details, I would pay more attention to those details.

Reply:
No worries. Here I believe we see this because `np.linalg` will run a single NumPy function over the array, while the `np.sum` variant will first create a whole array for the squared Jacobian before calling the NumPy sum function.